src/nncf/common/tensor_statistics/statistics.py

```diff
@@ -9,114 +9,300 @@
+        }
+
+    def load_data(self, loaded_data: dict[str, Tensor]) -> None:
+        self.mean_values = loaded_data[self.MEAN_STAT]
+        self.shape = tuple(loaded_data[self.SHAPE_STAT].tolist())
 
 
+@dataclass
 class MedianMADTensorStatistic(TensorStatistic):
-    MEDIAN_VALUES_STAT = "median_values"
-    MAD_VALUES_STAT = "mad_values"
+    MEDIAN_VALUES_STAT: ClassVar[str] = "median_values"
+    MAD_VALUES_STAT: ClassVar[str] = "mad_values"
 
-    def __init__(self, median_values: Tensor, mad_values: Tensor):
-        self.median_values = median_values
-        self.mad_values = mad_values
+    median_values: Tensor
+    mad_values: Tensor
 
     def __eq__(self, other: Any) -> bool:
-        if not isinstance(other, MedianMADTensorStatistic):
-            return False
-        return self.tensor_eq(self.median_values, other.median_values) and self.tensor_eq(
-            self.mad_values, other.mad_values
+        if isinstance(other, MedianMADTensorStatistic):
+            return fns.allclose(self.median_values, other.median_values) and fns.allclose(
+                self.mad_values, other.mad_values
+            )
+        return False
+
+    @classmethod
+    def from_config(cls, config: dict[str, Any]) -> TensorStatistic:
+        return cls(
+            median_values=config[cls.TENSOR_STATISTIC_OUTPUT_KEY][cls.MEDIAN_VALUES_STAT],
+            mad_values=config[cls.TENSOR_STATISTIC_OUTPUT_KEY][cls.MAD_VALUES_STAT],
         )
 
 
+@dataclass
 class PercentileTensorStatistic(TensorStatistic):
-    PERCENTILE_VS_VALUE_DICT = "percentile_vs_values_dict"
+    PERCENTILE_VS_VALUE_DICT: ClassVar[str] = "percentile_vs_values_dict"
 
-    def __init__(self, percentile_vs_values_dict: dict[float, Any]):
-        self.percentile_vs_values_dict = percentile_vs_values_dict
+    percentile_vs_values_dict: dict[int, Tensor]
 
-    def __eq__(self, other: Any, rtol: float = 1e-9) -> bool:
-        if not isinstance(other, PercentileTensorStatistic):
-            return False
-        if Counter(self.percentile_vs_values_dict.keys()) != Counter(other.percentile_vs_values_dict.keys()):
-            return False
-        for pct in self.percentile_vs_values_dict:
-            if not self.tensor_eq(self.percentile_vs_values_dict[pct], other.percentile_vs_values_dict[pct]):
+    def __eq__(self, other: Any) -> bool:
+        if isinstance(other, PercentileTensorStatistic):
+            if Counter(self.percentile_vs_values_dict.keys()) != Counter(other.percentile_vs_values_dict.keys()):
                 return False
-        return True
+            for pct in self.percentile_vs_values_dict:
+                if not fns.allclose(self.percentile_vs_values_dict[pct], other.percentile_vs_values_dict[pct]):
+                    return False
+            return True
+        return False
+
+    @classmethod
+    def from_config(cls, config: Union[dict[str, Any], dict[tuple[str, int], Any]]) -> TensorStatistic:
```

Comment:
Config can be Any? 
Looks like there is some problems with annotation of function.
Dont use cast instead of isinstance check.

Reply:
`config: dict[Any, Any]` ?